TalkAloud/TalkAloud/ViewControllers/AudioPlayerViewController.swift

```diff
@@ -46,11 +48,20 @@ class AudioPlayerViewController: UIViewController, AudioEngineStateChangeDelegat
             progressTimer?.invalidate()
             resetDurationLabels()
         }
+        
+        // Reset the visualizer in case we start playing
+        // another recording in the middle of playing another recording
+        audioPlayerVisualizer.waveforms.removeAll()
     }
     
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
         progressTimer?.invalidate()
+        visualizerTimer?.invalidate()
+        // Added this here to avoid additional complexity of timers when
+        // switching back between AudioRecordingsViewController and
+        // the AudioPlayerViewController
+        AudioEngine.sharedInstance.forceStopPlayer()
```

Comment:
Why not just call `stop` on the AudioEngine?

Reply:
Stop on AudioEngine doesn't stop playback which switching from the AuidoPlayerVIew to the AudioRecordingsView.